internal/errors/http/echo/default.go

```diff
@@ -0,0 +1,69 @@
+// package echo
+//
+// This package implements the default error handler to be
+// used for the api service to match the openapi specification.
+//
+// See: https://echo.labstack.com/docs/error-handling#custom-http-error-handler
+package echo
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+	"github.com/openlyinc/pointy"
+	api_public "github.com/podengo-project/idmsvc-backend/internal/api/public"
+	app_context "github.com/podengo-project/idmsvc-backend/internal/infrastructure/context"
+)
+
+func DefaultErrorHandler(err error, c echo.Context) {
+	logger := app_context.LogFromCtx(c.Request().Context())
+	if c.Response().Committed {
+		logger.Warn("response already commited",
+			slog.String("err", err.Error()),
+		)
+		return
+	}
+	errList := make([]api_public.ErrorInfo, 0, 10)
+
+	firstCode := 0
+	response := api_public.ErrorResponse{}
+	for {
+		logger.Debug("encapsulating error", slog.String("msg", err.Error()))
+		errInfo, code := parseError(err)
+		errList = append(errList, errInfo)
+		if firstCode == 0 {
+			firstCode = code
+		}
+		if err = errors.Unwrap(err); err == nil {
+			break
+		}
+	}
+	response.Errors = &errList
+	if err = c.JSON(firstCode, &response); err != nil {
+		logger.Error("the error could not be reported in the http response",
```

Comment:
Wouldn't message: "failed to send JSON error http response" be more precise here? I.e. "reported" is bit unusual thus confusing here. Also the fact that it failed in making it JSON is missing.